Automattic/woocommerce-payments
# Disable gateway when the currency is not supported in the country account

Fixes #2481 

#### Changes proposed in this Pull Request

Disable gateway when the currency is not supported in the country account

- Adds methods in the Payment Gateway to disable it if the selected currency is not supported by Stripe, using the `presentment_currencies` inside of Account (requires 992-gh-Automattic/woocommerce-payments-server)
- Adds a notice in the admin view if the currency is not part of `presentment_currencies`.

After #2496 has been merged this PR is mostly a fallback, as the currencies are already filtered out using the same `presentment_currencies` in the Multi Currency selector.

#### Testing instructions





* Make [this method](https://github.com/Automattic/woocommerce-payments/blob/95ee60b6b7944be7909fa711593c4c654bef901e/includes/class-wc-payment-gateway-wcpay.php#L535) return false
* Go to checkout and see the payment form disabled
* Go to Payments settings and see the notice text.

-------------------
